solo-chains/node/tanssi-relay-service/src/lib.rs

```diff
@@ -14,6 +14,602 @@
 // You should have received a copy of the GNU General Public License
 // along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
 
-//! Polkadot service. Specialized wrapper over substrate service.
+//! Development Polkadot service. Adapted from `polkadot_service` crate
+//! and removed un-necessary components which are not required in dev node.
+//!
+//! Following major changes are made:
+//! 1. Removed beefy and grandpa notification service and request response protocols
+//! 2. Removed support for parachains which also eliminated the need to start overseer and all other subsystems associated with collation + network request/response protocols for the same
+//! 3. Removed support for hardware benchmarking
+//! 4. Removed authority discovery service
+//! 5. Removed spawning of beefy, grandpa and MMR worker
+//! 6. Removed rpc extensions for beefy, grandpa and babe and added support for manual seal
+//! 7. Removed beefy and grandpa block import from block import pipeline (Babe remains)
+//! 8. Using manual seal import queue instead of babe import queue
+//! 9. Started manual seal worker
+
+use async_io::Timer;
+use babe::{BabeBlockImport, BabeLink};
+use consensus_common::SelectChain;
+use futures::FutureExt;
+use futures::Stream;
+use futures::StreamExt;
+use jsonrpsee::RpcModule;
+use node_common::service::Sealing;
+use polkadot_core_primitives::{AccountId, Balance, Block, BlockNumber, Hash, Nonce};
+use polkadot_node_core_parachains_inherent::Error as InherentError;
+use polkadot_overseer::Handle;
+use polkadot_primitives::InherentData as ParachainsInherentData;
+use polkadot_rpc::{DenyUnsafe, RpcExtension};
+use polkadot_service::{
+    BlockT, Error, IdentifyVariant, NewFullParams, OverseerGen, SelectRelayChain,
+};
+use sc_client_api::{AuxStore, Backend};
+use sc_consensus_manual_seal::consensus::babe::BabeConsensusDataProvider;
+use sc_consensus_manual_seal::rpc::{ManualSeal, ManualSealApiServer};
+use sc_consensus_manual_seal::{run_manual_seal, EngineCommand, ManualSealParams};
+use sc_executor::{HeapAllocStrategy, WasmExecutor, DEFAULT_HEAP_ALLOC_STRATEGY};
+use sc_transaction_pool_api::{OffchainTransactionPoolFactory, TransactionPool};
+use service::{Configuration, KeystoreContainer, RpcHandlers, TaskManager};
+use sp_api::ProvideRuntimeApi;
+use sp_block_builder::BlockBuilder;
+use sp_blockchain::{HeaderBackend, HeaderMetadata};
+use sp_consensus_babe::SlotDuration;
+use starlight_runtime::RuntimeApi;
+use std::sync::Arc;
+use std::time::Duration;
+use telemetry::{Telemetry, TelemetryWorker, TelemetryWorkerHandle};
 
 pub mod chain_spec;
+
+pub type FullBackend = service::TFullBackend<Block>;
+
+pub type FullClient = service::TFullClient<
+    Block,
+    RuntimeApi,
+    WasmExecutor<(
+        sp_io::SubstrateHostFunctions,
+        frame_benchmarking::benchmarking::HostFunctions,
+    )>,
+>;
+
+pub struct NewFull {
```

Comment:
In order to have clarity over this, I would call all this NewDevFull, `create_dev_rpc_extension`, `build_dev_full`. I would even call this file dev_service.rs, to make it clear that we only modify the dev-service and not the actual service

Reply:
I have moved it into `dev_service.rs` so it is now in a separate module itself. I did not rename it as it is redundant to say `dev_service::build_dev_full`.